src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesView/ExternalDataSourcesView.js

```diff
@@ -173,109 +202,122 @@ const ExternalDataSourcesView = ({
-                value={<FormattedMessage id={value.supportsHarvesting ? 'ui-local-kb-admin.yes' : 'ui-local-kb-admin.no'} />}
-              />
-            </Col>
-            <Col xs={4}>
-              <KeyValue
-                data-test-external-data-source-activationenabled
-                label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.activationEnabled" />}
-                value={<FormattedMessage id={value.activationEnabled ? 'ui-local-kb-admin.yes' : 'ui-local-kb-admin.no'} />}
-              />
-            </Col>
-          </Row>
-        </Layout>
         <Row>
           <Col xs={3}>
             <KeyValue
               data-test-external-data-source-listprefix
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.listPrefix" />}
-              value={value?.listPrefix ?? <NoValue />}
+              value={externalDataSource?.listPrefix ?? <NoValue />}
             />
           </Col>
           <Col xs={5}>
             <KeyValue
               data-test-external-data-source-fullprefix
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.fullPrefix" />}
-              value={value?.fullPrefix ?? <NoValue />}
+              value={externalDataSource?.fullPrefix ?? <NoValue />}
             />
           </Col>
           <Col xs={4}>
             <KeyValue
               data-test-external-data-source-principal
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.principal" />}
-              value={value?.principal ?? <NoValue />}
+              value={externalDataSource?.principal ?? <NoValue />}
             />
           </Col>
         </Row>
         <KeyValue
           data-test-external-data-source-credentials
           label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.credentials" />}
-          value={value?.credentials ?? <NoValue />}
+          value={externalDataSource?.credentials ?? <NoValue />}
+        />
+      </Pane>
+      {deleteModal && (
+        <ConfirmationModal
+          buttonStyle="danger"
+          confirmLabel={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.delete.confirmLabel" />}
+          data-test-confirmationModal
+          heading={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.delete.confirmHeading" />}
+          id="delete-external-data-source-confirmation"
+          message={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.delete.confirmMessage" values={{ name: externalDataSource?.name }} />}
+          onCancel={() => setDeleteModal(false)}
+          onConfirm={() => {
+            onDelete(externalDataSource?.id);
+            onClose();
+            setDeleteModal(false);
+          }}
+          open={deleteModal}
         />
-      </Card>
+      )}
+      <ExternalDataSourcesFormModal
+        initialValues={mode === EDITING ?
```

Comment:
Because we've split them out, we now only need the initial values corresponding to "EDIT", so you can remove this if/else ternary and just use the external data source values.

Importantly, this works because we're no longer passing external data source as state, instead it comes from the lookup, so good job! :D This code would almost certainly work as expected, I'm just asking for some minor cleanup